[Side Nav] Styling fixes for panel openers

tsullivan opened the pull request:
## Summary

Epic: https://github.com/elastic/kibana-team/issues/1439

For "panel opener" type of nav items:
1. Allow buttons for "panel opener" nav items to show the icon.
2. Adjust the size of the arrow for panel opener
3. Flips the direction the arrow points when the secondary nav is expanded

For all types of nav items:
1. Fix highlighted/hover colors

Other changes:
1. Refactored Emotion code to separate "static" and "dynamic" CSS rules.

**Not** addressed:
1. Long nav item title text that needs truncation and alignment fixes
2. Title text should never be underlined

### Screenshots

#### Icon fixes
**Before:** "Item 03", "Item 05", and "Item 06" were not showing the icon to the left of the title. The "Item 06" is showing an incorrect background color for the highlighted state.
<img width="826" alt="image" src="https://github.com/user-attachments/assets/8b5b57b7-398c-4a7e-9c93-ad549efd76f7" />

**After:** The icons are fixed, and the highlighted state color is fixed
<img width="833" alt="image" src="https://github.com/user-attachments/assets/94200dd9-b860-4e5d-9bec-ff9d0698a9e1" />

#### Panel fixes
The background color of the "selected" nav item in the panel is fixed. **More panel styling fixes are coming soon!**
<img width="826" alt="image" src="https://github.com/user-attachments/assets/a9588f3c-8a65-40c0-90fb-5d55de0ceb59" />

tsullivan committed:
Highlight/Hover color updates

tsullivan committed:
Icon stuff

tsullivan committed:
Color stuff

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293975)
* Commit: 877c835a006398a488f0603e09324abac22fb9b4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-218513)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/31f0f211d9d76fc757adaff91e41f94bafbacfb4) | [after](https://github.com/elastic/kibana/commit/877c835a006398a488f0603e09324abac22fb9b4) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.6KB | 18.9KB | +363.0B |
| `serverless` | 19.3KB | 19.7KB | +364.0B |
| total |  |  | +727.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/31f0f211d9d76fc757adaff91e41f94bafbacfb4) | [after](https://github.com/elastic/kibana/commit/877c835a006398a488f0603e09324abac22fb9b4) | diff |
| --- | --- | --- | --- |
| `@kbn/core-chrome-browser` | 211 | 212 | +1 |


</details>


<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":293975}
buildkite-pr-comment-->

tsullivan committed:
New folder for navigation_section components

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/294353)
* Commit: 4feb1a3c265725c380019295d2d46ed1b64cddd1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/31f0f211d9d76fc757adaff91e41f94bafbacfb4) | [after](https://github.com/elastic/kibana/commit/4feb1a3c265725c380019295d2d46ed1b64cddd1) | diff |
| --- | --- | --- | --- |
| `navigation` | 48 | 49 | +1 |
| `serverless` | 43 | 44 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/31f0f211d9d76fc757adaff91e41f94bafbacfb4) | [after](https://github.com/elastic/kibana/commit/4feb1a3c265725c380019295d2d46ed1b64cddd1) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.6KB | 19.4KB | +833.0B |
| `serverless` | 19.3KB | 20.1KB | +836.0B |
| total |  |  | +1.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/31f0f211d9d76fc757adaff91e41f94bafbacfb4) | [after](https://github.com/elastic/kibana/commit/4feb1a3c265725c380019295d2d46ed1b64cddd1) | diff |
| --- | --- | --- | --- |
| `navigation` | 12.7KB | 12.7KB | -2.0B |



### History
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":294353}
buildkite-pr-comment-->

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
Merge branch 'sidenav/fix-icons-for-panelopeners' of github.com:tsullivan/kibana into sidenav/fix-icons-for-panelopeners

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/294719)
* Commit: da3c57e3217f1573f7ff0d36f09138ea277899cf

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/4e6fa9e37f7390a86b0f38f8981fbf9eb1f0751b) | [after](https://github.com/elastic/kibana/commit/da3c57e3217f1573f7ff0d36f09138ea277899cf) | diff |
| --- | --- | --- | --- |
| `navigation` | 48 | 49 | +1 |
| `serverless` | 44 | 45 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/4e6fa9e37f7390a86b0f38f8981fbf9eb1f0751b) | [after](https://github.com/elastic/kibana/commit/da3c57e3217f1573f7ff0d36f09138ea277899cf) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.6KB | 19.4KB | +833.0B |
| `serverless` | 19.3KB | 20.1KB | +836.0B |
| total |  |  | +1.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/4e6fa9e37f7390a86b0f38f8981fbf9eb1f0751b) | [after](https://github.com/elastic/kibana/commit/da3c57e3217f1573f7ff0d36f09138ea277899cf) | diff |
| --- | --- | --- | --- |
| `navigation` | 12.7KB | 12.7KB | -2.0B |



### History
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":294719}
buildkite-pr-comment-->

ek-so reviewed (commented):
I'd not flip the arrow — it looks fine when the arrow is placed horizontally, like an accordion, but in this case tbh looks a little bit confusing. (I believe, it was not mentioned in [requirements](https://github.com/user-attachments/assets/853a8fc9-fffe-4619-982f-c564d8644a94)?)

tsullivan committed:
fix "expanded" panel opener bg color style

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/295143)
* Commit: 79d6bfe75e9059652309c632b7b5fbaf17d28bd0

### Failed CI Steps
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/295143#01965f75-0480-475f-b716-76318ebe6ad8)
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/295143#01965f87-41d8-44d3-91fc-97c32f916a27)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295143#01965f75-0480-475f-b716-76318ebe6ad8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295143/jobs/01965f75-0480-475f-b716-76318ebe6ad8/artifacts/01965f87-1650-4e16-a725-4e2c96ea5d1f) FTR Configs #<span></span>112 / serverless observability UI navigation navigate observability sidenav & breadcrumbs
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295143#01965f87-41d8-44d3-91fc-97c32f916a27) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295143/jobs/01965f87-41d8-44d3-91fc-97c32f916a27/artifacts/01965f90-5f76-4e40-a634-abf3b0cd5b95) FTR Configs #<span></span>112 / serverless observability UI navigation navigate observability sidenav & breadcrumbs

### History
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":295143}
buildkite-pr-comment-->

tsullivan committed:
fix selected color in nav panel

tsullivan committed:
Fix tests

tsullivan committed:
Minor cleanup

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/295170)
* Commit: 65f248d7be53caacfe876bd12c666b30b4b503d9

### Failed CI Steps
* [FTR Configs #8](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01965fe3-e91f-41dc-b494-2594055c60ee)
* [FTR Configs #8](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01966008-2a5a-4199-b7b2-c25818cefcdf)
* [FTR Configs #111](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01965fe3-e96c-4255-851d-46798b0c6545)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01965fe3-e91f-41dc-b494-2594055c60ee) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295170/jobs/01965fe3-e91f-41dc-b494-2594055c60ee/artifacts/01966008-0037-4bde-b53f-a812acbddc99) FTR Configs #<span></span>8 / Solution navigation smoke tests security solution sidenav & breadcrumbs renders the correct nav and navigate to links
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01966008-2a5a-4199-b7b2-c25818cefcdf) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295170/jobs/01966008-2a5a-4199-b7b2-c25818cefcdf/artifacts/0196600e-e7a3-42fc-af0a-8c373977d5a4) FTR Configs #<span></span>8 / Solution navigation smoke tests security solution sidenav & breadcrumbs renders the correct nav and navigate to links
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295170#01965fe3-e96c-4255-851d-46798b0c6545) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295170/jobs/01965fe3-e96c-4255-851d-46798b0c6545/artifacts/01966002-8375-4850-8316-798381e8f08c) FTR Configs #<span></span>111 / Stateful Observability - Deployment-agnostic API integration tests SyntheticsAPITests SyntheticsEnablement [PUT] /internal/uptime/service/enablement returns response for an admin with privilege

### History
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":295170}
buildkite-pr-comment-->

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
fix padding of panel opener w/out icon

tsullivan committed:
fix tests ii

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/295568)
* Commit: 5e02ae8509e5c735769e1fe4c7b81ed2f8078030

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/5e02ae8509e5c735769e1fe4c7b81ed2f8078030) | diff |
| --- | --- | --- | --- |
| `navigation` | 44 | 45 | +1 |
| `serverless` | 40 | 41 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/5e02ae8509e5c735769e1fe4c7b81ed2f8078030) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.6KB | 19.5KB | +918.0B |
| `serverless` | 19.3KB | 20.2KB | +921.0B |
| total |  |  | +1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/5e02ae8509e5c735769e1fe4c7b81ed2f8078030) | diff |
| --- | --- | --- | --- |
| `navigation` | 11.9KB | 11.9KB | -2.0B |



### History
* :broken_heart: [Build #295170](https://buildkite.com/elastic/kibana-pull-request/builds/295170) failed 65f248d7be53caacfe876bd12c666b30b4b503d9
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295568","number":295568,"commit":"5e02ae8509e5c735769e1fe4c7b81ed2f8078030"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293975","number":293975,"commit":"877c835a006398a488f0603e09324abac22fb9b4"}],"number":295568}
buildkite-pr-comment-->

tsullivan committed:
Fix spacing of button and divider

tsullivan committed:
Fix gap issue 2: for items with icons

tsullivan commented:
@Dosant I've fixed the gap problem with the "panel opener" buttons
<img width="278" alt="image" src="https://github.com/user-attachments/assets/50fbd919-3f20-4440-bd83-69a7d00d1502" />

@ek-so I've fixed the issue you found in my next PR, here: https://github.com/elastic/kibana/pull/218050#pullrequestreview-2783544842. It is visible in the storybook of this PR.
<img width="875" alt="image" src="https://github.com/user-attachments/assets/97ba0d1a-3ee6-4f48-9407-0ea533e9d260" />

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/295598)
* Commit: f2a5c5263a2a56584365ddd8ccc121323fc53085

### Failed CI Steps
* [FTR Configs #86](https://buildkite.com/elastic/kibana-pull-request/builds/295598#0196655c-39f0-484b-a5b4-89a38b959452)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/295598#0196655c-39f0-484b-a5b4-89a38b959452) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/295598/jobs/0196655c-39f0-484b-a5b4-89a38b959452/artifacts/01966584-3bd6-4fb0-85d4-0fa5e145acec) FTR Configs #<span></span>86 / disable Elastic Maps Service Fonts "before all" hook for "should load map with labels"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/f2a5c5263a2a56584365ddd8ccc121323fc53085) | diff |
| --- | --- | --- | --- |
| `navigation` | 44 | 45 | +1 |
| `serverless` | 40 | 41 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/f2a5c5263a2a56584365ddd8ccc121323fc53085) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.6KB | 19.7KB | +1.1KB |
| `serverless` | 19.3KB | 20.5KB | +1.1KB |
| total |  |  | +2.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f15c96d6e811c7a1c1b206c1a0b98c24187ddecf) | [after](https://github.com/elastic/kibana/commit/f2a5c5263a2a56584365ddd8ccc121323fc53085) | diff |
| --- | --- | --- | --- |
| `navigation` | 11.9KB | 11.9KB | -2.0B |



### History
* :green_heart: [Build #295568](https://buildkite.com/elastic/kibana-pull-request/builds/295568) succeeded 5e02ae8509e5c735769e1fe4c7b81ed2f8078030
* :broken_heart: [Build #295170](https://buildkite.com/elastic/kibana-pull-request/builds/295170) failed 65f248d7be53caacfe876bd12c666b30b4b503d9
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1
* :green_heart: [Build #293975](https://buildkite.com/elastic/kibana-pull-request/builds/293975) succeeded 877c835a006398a488f0603e09324abac22fb9b4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295598","number":295598,"commit":"f2a5c5263a2a56584365ddd8ccc121323fc53085"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295568","number":295568,"commit":"5e02ae8509e5c735769e1fe4c7b81ed2f8078030"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294353","number":294353,"commit":"4feb1a3c265725c380019295d2d46ed1b64cddd1"}],"number":295598}
buildkite-pr-comment-->

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
Optimize emotion code

tsullivan committed:
naming tweak

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296420)
* Commit: debc4f61a5f12c3414f319c84b3de4292c8314bb
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-218513)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/82c13c9b2a96f139364263b57a88eee3a4bafaf3) | [after](https://github.com/elastic/kibana/commit/debc4f61a5f12c3414f319c84b3de4292c8314bb) | diff |
| --- | --- | --- | --- |
| `navigation` | 44 | 45 | +1 |
| `serverless` | 40 | 41 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/82c13c9b2a96f139364263b57a88eee3a4bafaf3) | [after](https://github.com/elastic/kibana/commit/debc4f61a5f12c3414f319c84b3de4292c8314bb) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.7KB | 20.6KB | +1.9KB |
| `serverless` | 19.4KB | 21.3KB | +1.9KB |
| total |  |  | +3.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/82c13c9b2a96f139364263b57a88eee3a4bafaf3) | [after](https://github.com/elastic/kibana/commit/debc4f61a5f12c3414f319c84b3de4292c8314bb) | diff |
| --- | --- | --- | --- |
| `navigation` | 11.9KB | 11.9KB | -2.0B |



### History
* :yellow_heart: [Build #295598](https://buildkite.com/elastic/kibana-pull-request/builds/295598) was flaky f2a5c5263a2a56584365ddd8ccc121323fc53085
* :green_heart: [Build #295568](https://buildkite.com/elastic/kibana-pull-request/builds/295568) succeeded 5e02ae8509e5c735769e1fe4c7b81ed2f8078030
* :broken_heart: [Build #295170](https://buildkite.com/elastic/kibana-pull-request/builds/295170) failed 65f248d7be53caacfe876bd12c666b30b4b503d9
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf
* :green_heart: [Build #294353](https://buildkite.com/elastic/kibana-pull-request/builds/294353) succeeded 4feb1a3c265725c380019295d2d46ed1b64cddd1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296420","number":296420,"commit":"debc4f61a5f12c3414f319c84b3de4292c8314bb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295598","number":295598,"commit":"f2a5c5263a2a56584365ddd8ccc121323fc53085"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295568","number":295568,"commit":"5e02ae8509e5c735769e1fe4c7b81ed2f8078030"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294719","number":294719,"commit":"da3c57e3217f1573f7ff0d36f09138ea277899cf"}],"number":296420}
buildkite-pr-comment-->

eokoneyo reviewed (commented):
Tested this out in storybook, there's some things I noticed;

I might be mistaken but I think this standalone menu item should line up with the text and icon start, also it protrudes out of  it's own hover highlight.
<img width="255" alt="Screenshot 2025-04-26 at 00 18 09" src="https://github.com/user-attachments/assets/d1d6f75f-245d-43ca-b0d4-50ebdaac4d66" />

I also noticed that the nested menu item here, exceeds the left margin of it's parent and isn't aligned with it's siblings
<img width="266" alt="Screenshot 2025-04-26 at 00 16 56" src="https://github.com/user-attachments/assets/f6366b06-2eb8-4da8-bb62-5c0365770e98" />

tsullivan committed:
simplify css rules, fix euiCollapsibleNavLink

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

tsullivan committed:
refine spacing of panel opener button and icon

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296838)
* Commit: e1113181876f94664f90569e13dc675f53a67a01
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-218513)

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/296838#01967d7b-b296-4fdf-a131-463401f150b3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/296838#01967d7b-b296-4fdf-a131-463401f150b3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/296838/jobs/01967d7b-b296-4fdf-a131-463401f150b3/artifacts/01967da3-87df-408c-a51e-7f3695cdb6ca) Jest Tests #<span></span>1 / TableTabTour should render

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ca5cae90f9ab30eab0a52cfb785ef01d76be9a7d) | [after](https://github.com/elastic/kibana/commit/e1113181876f94664f90569e13dc675f53a67a01) | diff |
| --- | --- | --- | --- |
| `navigation` | 44 | 45 | +1 |
| `serverless` | 40 | 41 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ca5cae90f9ab30eab0a52cfb785ef01d76be9a7d) | [after](https://github.com/elastic/kibana/commit/e1113181876f94664f90569e13dc675f53a67a01) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.7KB | 20.3KB | +1.6KB |
| `serverless` | 19.4KB | 21.0KB | +1.6KB |
| total |  |  | +3.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ca5cae90f9ab30eab0a52cfb785ef01d76be9a7d) | [after](https://github.com/elastic/kibana/commit/e1113181876f94664f90569e13dc675f53a67a01) | diff |
| --- | --- | --- | --- |
| `navigation` | 11.9KB | 11.9KB | -2.0B |



### History
* :green_heart: [Build #296420](https://buildkite.com/elastic/kibana-pull-request/builds/296420) succeeded debc4f61a5f12c3414f319c84b3de4292c8314bb
* :yellow_heart: [Build #295598](https://buildkite.com/elastic/kibana-pull-request/builds/295598) was flaky f2a5c5263a2a56584365ddd8ccc121323fc53085
* :green_heart: [Build #295568](https://buildkite.com/elastic/kibana-pull-request/builds/295568) succeeded 5e02ae8509e5c735769e1fe4c7b81ed2f8078030
* :broken_heart: [Build #295170](https://buildkite.com/elastic/kibana-pull-request/builds/295170) failed 65f248d7be53caacfe876bd12c666b30b4b503d9
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0
* :green_heart: [Build #294719](https://buildkite.com/elastic/kibana-pull-request/builds/294719) succeeded da3c57e3217f1573f7ff0d36f09138ea277899cf

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296838","number":296838,"commit":"e1113181876f94664f90569e13dc675f53a67a01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296420","number":296420,"commit":"debc4f61a5f12c3414f319c84b3de4292c8314bb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295598","number":295598,"commit":"f2a5c5263a2a56584365ddd8ccc121323fc53085"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295568","number":295568,"commit":"5e02ae8509e5c735769e1fe4c7b81ed2f8078030"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295143","number":295143,"commit":"79d6bfe75e9059652309c632b7b5fbaf17d28bd0"}],"number":296838}
buildkite-pr-comment-->

tsullivan committed:
Merge branch 'main' into sidenav/fix-icons-for-panelopeners

ryankeairns commented:
@tsullivan should any of these custom styles be applied at the EUI level? (e.g. `EuiCollapsibleNavItem`)

Not asking to block the PR, but I am thinking some of these should ultimately live there and, once added, we could remove the custom that get merged in here. If so, the process would be to open an issue in EUI and reference the custom styles from this PR.

ek-so reviewed (approved):
As we discussed — a couple of things will get together in another PR to make a whole picture

tsullivan commented:
> > @tsullivan should any of these custom styles be applied at the EUI level? (e.g. EuiCollapsibleNavItem)

> Not asking to block the PR, but I am thinking some of these should ultimately live there and, once added, we could remove the custom styles that get merged in here. If so, the process would be to open an issue in EUI and reference the custom styles from this PR.

Hi @ryankeairns, this is an excellent consideration. Many of the updates can't be applied at the EUI level: the "panel openers" are a customization that lives in Kibana and is built "on top" of the EUI component. But there are also changes to accordions and regular nav items that are part of the EUI component. I will file an issue to place the appropriate styles in the EUI repo and remove the customizations from this Kibana package. Thank you!

tsullivan commented:
> As we discussed — a couple of things will get together in another PR to make a whole picture

This refers to: https://github.com/elastic/kibana/pull/218050. Parts of the footer have invalid styles because footer items currently have an invalid structure.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/297618)
* Commit: b578ed6969fed9d91e2ae99720b0c5b1cbf3d737
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-218513)

### Failed CI Steps
* [FTR Configs #67](https://buildkite.com/elastic/kibana-pull-request/builds/297618#01968718-b977-4d0f-911b-dcb0ca6ccd69)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ec008b0a63bb83b70cdc898f7571c5f52aee66e8) | [after](https://github.com/elastic/kibana/commit/b578ed6969fed9d91e2ae99720b0c5b1cbf3d737) | diff |
| --- | --- | --- | --- |
| `navigation` | 44 | 45 | +1 |
| `serverless` | 40 | 41 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ec008b0a63bb83b70cdc898f7571c5f52aee66e8) | [after](https://github.com/elastic/kibana/commit/b578ed6969fed9d91e2ae99720b0c5b1cbf3d737) | diff |
| --- | --- | --- | --- |
| `navigation` | 18.7KB | 20.3KB | +1.6KB |
| `serverless` | 19.4KB | 21.0KB | +1.6KB |
| total |  |  | +3.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ec008b0a63bb83b70cdc898f7571c5f52aee66e8) | [after](https://github.com/elastic/kibana/commit/b578ed6969fed9d91e2ae99720b0c5b1cbf3d737) | diff |
| --- | --- | --- | --- |
| `navigation` | 11.9KB | 11.9KB | -2.0B |



### History
* :yellow_heart: [Build #296838](https://buildkite.com/elastic/kibana-pull-request/builds/296838) was flaky e1113181876f94664f90569e13dc675f53a67a01
* :green_heart: [Build #296420](https://buildkite.com/elastic/kibana-pull-request/builds/296420) succeeded debc4f61a5f12c3414f319c84b3de4292c8314bb
* :yellow_heart: [Build #295598](https://buildkite.com/elastic/kibana-pull-request/builds/295598) was flaky f2a5c5263a2a56584365ddd8ccc121323fc53085
* :green_heart: [Build #295568](https://buildkite.com/elastic/kibana-pull-request/builds/295568) succeeded 5e02ae8509e5c735769e1fe4c7b81ed2f8078030
* :broken_heart: [Build #295170](https://buildkite.com/elastic/kibana-pull-request/builds/295170) failed 65f248d7be53caacfe876bd12c666b30b4b503d9
* :broken_heart: [Build #295143](https://buildkite.com/elastic/kibana-pull-request/builds/295143) failed 79d6bfe75e9059652309c632b7b5fbaf17d28bd0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/297618","number":297618,"commit":"b578ed6969fed9d91e2ae99720b0c5b1cbf3d737"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296838","number":296838,"commit":"e1113181876f94664f90569e13dc675f53a67a01"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296420","number":296420,"commit":"debc4f61a5f12c3414f319c84b3de4292c8314bb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295598","number":295598,"commit":"f2a5c5263a2a56584365ddd8ccc121323fc53085"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295568","number":295568,"commit":"5e02ae8509e5c735769e1fe4c7b81ed2f8078030"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/295170","number":295170,"commit":"65f248d7be53caacfe876bd12c666b30b4b503d9"}],"number":297618}
buildkite-pr-comment-->
